Sources/Rendering/OpenGL/ShaderProgram/index.js

```diff
@@ -184,6 +184,26 @@ export function vtkShaderProgram(publicAPI, model) {
     return true;
   };
 
+  publicAPI.setUniform4f = (name, v) => {
+    const location = publicAPI.findUniform(name);
+    if (location === -1) {
+      model.error = `Could not set uniform ${name} . No such uniform.`;
+      return false;
+    }
+    model.context.uniform4f(location, v[0], v[1], v[2], v[3]);
+    return true;
+  };
+
+  publicAPI.setUniform4fv = (name, count, v) => {
+    const location = publicAPI.findUniform(name);
+    if (location === -1) {
+      model.error = `Could not set uniform ${name} . No such uniform.`;
+      return false;
+    }
+    model.context.uniform4fv(location, count, v);
```

Comment:
Yes, count should be removed from the signatures.

Here is an example of what I think it should be, I did not test it but that should give you the idea.

 publicAPI.setUniform3f = (name, ...args) => {
    const location = publicAPI.findUniform(name);
    if (location === -1) {
      model.error = `Could not set uniform ${name} . No such uniform.`;
      return false;
    }

    let array = args;
    // allow an array passed as a single arg.
    if (array.length === 1 && Array.isArray(array[0])) {
      array = array[0];
    }

    if (array.length !== 3) {
      throw new RangeError('Invalid number of values for array');
    }

    model.context.uniform3f(location, array[0], array[1], array[2]);
    return true;
  };

  publicAPI.setUniform3fv = (name, v) => {
    const location = publicAPI.findUniform(name);
    if (location === -1) {
      model.error = `Could not set uniform ${name} . No such uniform.`;
      return false;
    }
    model.context.uniform3fv(location, v);
    return true;
  };

FYI the difference in the 4f versus 4fv is that 4fv will take any multiple of 4 args. So you can do

vec4 colors[5];

and then setUniform4fv(colors, array[20])

and it will set 5 (which was count)  vec4s at once.